packages/runtime/src/runtime/menu.tsx

```diff
@@ -130,38 +159,239 @@ const CustomLink: React.FC<PropsWithChildren & { item: MenuItem }> = ({
+        values={values}
+      />
+    );
+  }
+  if (type === "Drawer") {
+    return (
+      <DrawerMenu
+        handleClose={handleClose}
+        isOpen={isCollapsed}
+        selectedItem={selectedItem}
+        setSelectedItem={setSelectedItem}
+        values={values}
+      />
+    );
+  }
+  return null;
+};
 
+export const SideBarMenu: React.FC<{
+  values?: MenuBaseProps<SideBarMenuStyles>;
+  isCollapsed: boolean;
+  selectedItem: string | undefined;
+  setSelectedItem: (s: string) => void;
+  width?: string;
+}> = ({ values, isCollapsed, selectedItem, setSelectedItem }) => {
+  return (
+    <Col
+      id="ensemble-sidebar"
+      style={{
+        backgroundColor: values?.styles?.backgroundColor
+          ? getColor(values.styles.backgroundColor)
+          : "none",
+        borderRight: "1px solid lightgrey",
+        display: "flex",
+        justifyContent: "space-between",
+        flexDirection: "column",
+        width: isCollapsed ? undefined : values?.styles?.width,
+        height: "100vh",
+        alignItems: "center",
+        zIndex: 1,
+      }}
+    >
+      {values?.header ? EnsembleRuntime.render([values.header]) : null}
+      <ItemsMenu
+        isCollapsed={isCollapsed}
+        items={values?.items || []}
+        selectedItem={selectedItem}
+        setSelectedItem={setSelectedItem}
+        styles={values?.styles || {}}
+      />
+      {values?.footer ? EnsembleRuntime.render([values.footer]) : null}
+    </Col>
+  );
+};
+
+export const DrawerMenu: React.FC<{
+  values?: MenuBaseProps<DrawerMenuStyles>;
+  handleClose: () => void;
+  isOpen: boolean;
+  selectedItem: string | undefined;
+  setSelectedItem: (s: string) => void;
+}> = ({ values, handleClose, isOpen, selectedItem, setSelectedItem }) => {
+  const validPosition = ["left", "right", "top", "bottom"];
+
+  return (
+    <AntDrawer
+      closable={false}
+      height={
+        values?.styles?.height
+          ? `calc(${values.styles.height} + 48px)`
+          : undefined
+      }
+      key={values?.id}
+      onClose={handleClose}
+      open={isOpen}
+      placement={
+        !(
+          values?.styles?.position &&
+          validPosition.includes(values.styles.position)
+        )
+          ? "left"
+          : values.styles.position
+      }
+      style={{
+        backgroundColor: values?.styles?.backgroundColor
+          ? getColor(values.styles.backgroundColor)
+          : "none",
+      }}
+      width={
+        values?.styles?.width
+          ? `calc(${values.styles.width} + 48px)`
+          : undefined
+      }
+    >
+      {values?.header ? EnsembleRuntime.render([values.header]) : null}
+      <ItemsMenu
+        items={values?.items || []}
+        selectedItem={selectedItem}
+        setSelectedItem={setSelectedItem}
+        styles={values?.styles || {}}
+      />
+      {values?.footer ? EnsembleRuntime.render([values.footer]) : null}
+    </AntDrawer>
+  );
+};
+
+const ItemsMenu: React.FC<{
```

Comment:
This is a good call breaking this out into its separate component. Can we rename to `MenuItems`?